crates/axevent/src/flex.rs

```diff
@@ -63,6 +63,39 @@ macro_rules! try_func {
     }}
 }
 
+#[repr(transparent)]
+pub struct CStringPtr(NonNull<c_char>);
+
+impl CStringPtr {
+    /// Create an owned string from a foreign allocation
+    ///
+    /// # Safety
+    ///
+    /// In addition to the safety preconditions for [`CStr::from_ptr`] the memory must have been
+    /// allocated in a manner compatible with [`glib_sys::g_free`] and there must be no other
+    /// users of this memory.
+    unsafe fn from_ptr(ptr: *mut c_char) -> Self {
+        Self(NonNull::new_unchecked(ptr))
```

Comment:
`ptr` needs to be checked for null or we introduce UB here. To me it seems preferable to panic here if ptr is null.

Reply:
This is a documented safety precondition for this function, just like `CStr::from_ptr` and I'd like to keep it that way to not force needless asserts into users' code. But I'm open to taking a page from `glib`s book and adding a debug assert if you think that would be an improvement: https://gtk-rs.org/gtk-rs-core/git/docs/src/glib/gstring.rs.html#2009